modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/ServerApiAccess.java

```diff
@@ -0,0 +1,327 @@
+				}
+			} else {
+				httpPost.setEntity(null);
+			}
+			Log.e("url", "" + url);
+			HttpPost httpPostWithHeaders = (HttpPost) buildHeaders(httpPost, headers, httpMethod);
+			try {
+				HttpResponse response = httpclient.execute(httpPostWithHeaders);
+				String status = String.valueOf(response.getStatusLine().getStatusCode());
+				Log.d(TAG, status);
+				responseParams.put("response", getResponseBody(response));
+				responseParams.put("status", status);
+				return responseParams;
+			} catch (ClientProtocolException e) {
+				Log.d(TAG, "ClientProtocolException :" + e.toString());
+				return null;
+			} catch (IOException e) {
+				Log.d(TAG, e.toString());
+				responseParams.put("response", "Internal Server Error");
+				responseParams.put("status", "500");
+				return responseParams;
+			}
+		} else if (httpMethod.equals("GET")) {
+			// if(payload!=null){
+			// url = url+"?"+payload;
+			// }
+			HttpGet httpGet = new HttpGet(url);
+			HttpGet httpGetWithHeaders = (HttpGet) buildHeaders(httpGet, headers, httpMethod);
+			Log.d(TAG, httpGetWithHeaders.toString() + " GET");
+			try {
+				HttpResponse response = httpclient.execute(httpGetWithHeaders);
+				responseParams.put("response", getResponseBody(response));
+				responseParams.put("status",
+				                   String.valueOf(response.getStatusLine().getStatusCode()));
+				return responseParams;
+			} catch (ClientProtocolException e) {
+				Log.d(TAG, "ClientProtocolException :" + e.toString());
+				return null;
+			} catch (IOException e) {
+				Log.d(TAG, e.toString());
+				responseParams.put("response", "Internal Server Error");
+				responseParams.put("status", "500");
+				return responseParams;
+			}
+		}
+		return null;
+	}
+
+	public static Map<String, String> postDataAPI(APIUtilities apiUtilities,
+	                                              Map<String, String> headers) {
+		String httpMethod = apiUtilities.getHttpMethod();
+		String url = apiUtilities.getEndPoint();
+		Map<String, String> params = apiUtilities.getRequestParamsMap();
+
+		Map<String, String> response_params = new HashMap<String, String>();
+		HttpClient httpclient = getCertifiedHttpClient();
+		String payload = buildPayload(params);
+
+		if (httpMethod.equals("POST")) {
+			HttpPost httpPost = new HttpPost(url);
+			Log.e("url", "" + url);
+			HttpPost httpPostWithHeaders = (HttpPost) buildHeaders(httpPost, headers, httpMethod);
+			byte[] postData = payload.getBytes();
+			try {
+				httpPostWithHeaders.setEntity(new ByteArrayEntity(postData));
+				HttpResponse response = httpclient.execute(httpPostWithHeaders);
+				String status = String.valueOf(response.getStatusLine().getStatusCode());
+				Log.d(TAG, status);
+				response_params.put("response", getResponseBody(response));
+				response_params.put("status", status);
+				return response_params;
+			} catch (ClientProtocolException e) {
+				Log.d(TAG, "ClientProtocolException :" + e.toString());
+				return null;
```

Comment:
Is there any specific reason why we need to catch exceptions and return a value? Can't we throw a proper wrapped exception upwards here in this particular instance?